packages/ngtools/webpack/src/ivy/plugin.ts

```diff
@@ -301,6 +299,26 @@ export class AngularWebpackPlugin {
     });
   }
 
+  private markResourceUsed(
+    normalizedResourcePath: string,
+    currentUnused: Set<string>,
+    visitedResources: Set<string>,
+  ): void {
+    if (visitedResources.has(normalizedResourcePath)) {
```

Comment:
Why do we need to keep track of `visitedResources` separately? We can use `currentUnused` as our single source of truth.

```ts
if (!currentUnused.has(normalizedResourcePath)) {
    return;
}

currentUnused.delete(normalizedResourcePath);

const dependencies = this.fileDependencies.get(normalizedResourcePath);
if (!dependencies) {
    return;
}

for (const dependency of dependencies) {
    this.markResourceUsed(normalizePath(dependency), currentUnused);
}
```

Reply:
I think Alan is correct; there is a risk for Webpack modules outside of the set of non-declaration source files, as cycles between those files wouldn't be detected through `currentUnused`, but this should not be a problem in practicle as `this.fileDependencies` only tracks deps between source files in the program.

Reply:
I tend to be wary of multi-purpose situations since a future change of one part may unintentionally break an invariant of the second.  But in this case both are localized and very specific in purpose so I think this makes sense.